index.js

```diff
@@ -3,6 +3,17 @@
 var has = Object.prototype.hasOwnProperty;
 
 /**
+ * Decode a URI encoded string.
+ *
+ * @param {String} input The URI encoded string.
+ * @returns {String} The decoded string.
+ * @api private
+ */
+function decode(input) {
+  return decodeURIComponent(input).replace(/\+/g, ' ');
```

Comment:
You have to replace before encoding, otherwise encoded plus signs get replace too:

```js
      var obj = qs.parse('foo+bar=baz%2Bqux');

      assume(obj).is.a('object');
      assume(obj['foo bar']).equals('baz+qux');
```